### PR TITLE
find_by_icao

### PR DESCRIPTION
**Pull request by ryanburnette**
This adds `#icao_codes` and `.find_by_icao_code` methods.

### CODE REVIEW

**Comment by timrogers** [comment]
Thanks for the PR! I’ll give this a review tomorrow.

File: lib/airports.rb [hunk 1]
```diff
@@ -11,10 +11,24 @@ def self.find_by_iata_code(iata_code)
       new(airport_data.each_with_object({}) { |(k,v), h| h[k.to_sym] = v })
   end
 
+  def self.find_by_icao_code(icao_code)
+    return unless icao_code.length == 4
```
Comment (timrogers) [inline lib/airports.rb]: I think I'd be inclined to take an approach more like this:

```ruby
def self.find_by_icao_code(icao_code)
  return unless icao_code.length == 4

  airport_data = parsed_data.find do |_, data|
    data['icao'] == icao_code
  end

  return unless airport_data

  Airport.
    new(airport_data.each_with_object({}) { |(k,v), h| h[k.to_sym] = v })
end
```

What do you think?

Comment (ryanburnette) [reply lib/airports.rb]: That looks like a more elegant solution. Wouldn't have thought about doing it that way.

**Comment by timrogers** [comment]
@ryanburnette Do you want to make that change, and then we can get this merged?

**Comment by ryanburnette** [comment]
Sure

**Comment by ryanburnette** [comment]
I wasn't able to get the snippet you provided to work.

**Comment by timrogers** [comment]
I've released this feature in v1.1.0 - nice work ❤️ What are you planning to use it for?

**Comment by timrogers** [comment]
I fixed and merged this code in 69ccb1a4adf02fe9e5606a29c05a8097841a0717.

**Comment by ryanburnette** [comment]
https://github.com/ryanburnette/vfrmap

**Comment by ryanburnette** [comment]
And be careful to put the "e" on the end of my name. That Ryan Burnett has gotten more credit for my work that I'd like to admit. 😂I'm Ryan Burnette.

**Comment by ryanburnette** [comment]
And hey, just out of curiosity, was your goal with the code you used to be more performance efficient? I'm not a Ruby expert by any means ... I generally write things so they work and are easy to read for me.

**Comment by timrogers** [comment]
I've fixed your name - sorry about that!

It's a mix of both really - it'll be more performant and feels like more idiomatic Ruby.

**Pull request closed without merging**